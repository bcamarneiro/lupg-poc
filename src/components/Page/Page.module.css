.page {
  @apply max-w-screen-md m-auto py-9;
}
