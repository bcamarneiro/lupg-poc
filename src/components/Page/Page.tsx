import { PropsWithChildren } from 'react';
import style from './Page.module.css';

function Page({ children }: PropsWithChildren) {
  return <main className={style.page}>{children}</main>;
}
export default Page;
