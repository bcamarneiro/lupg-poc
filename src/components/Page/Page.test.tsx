import { render, screen } from '@testing-library/react';
import Page from './Page';

describe('Page component', () => {
  it('renders children correctly', () => {
    const { asFragment } = render(
      <Page>
        <div>Test Content</div>
      </Page>,
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText('Test Content')).toBeInTheDocument();
  });

  it('applies correct CSS class', () => {
    render(
      <Page>
        <div>Test Content</div>
      </Page>,
    );
    const mainElement = screen.getByRole('main');
    expect(mainElement.className).contains('page');
  });

  it('renders multiple children', () => {
    render(
      <Page>
        <div>Child 1</div>
        <div>Child 2</div>
        <div>Child 3</div>
      </Page>,
    );
    expect(screen.getByText('Child 1')).toBeInTheDocument();
    expect(screen.getByText('Child 2')).toBeInTheDocument();
    expect(screen.getByText('Child 3')).toBeInTheDocument();
  });
});
