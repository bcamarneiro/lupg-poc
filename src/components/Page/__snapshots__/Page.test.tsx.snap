// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Page component > renders children correctly 1`] = `
<DocumentFragment>
  <main
    class="_page_e95033"
  >
    <div>
      Test Content
    </div>
  </main>
</DocumentFragment>
`;
